### PR TITLE
Repair solar_azimuth_analytical

### PR DESCRIPTION
**Pull request by adriesse**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python!

You may submit a pull request with your code at any stage of completion, however, before the code can be merged the following items must be addressed:

 - [x] Closes issue #420 
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests must pass on the TravisCI and Appveyor testing services.
 - [x] Code quality and style is sufficient. Passes ``git diff upstream/master -u -- "*.py" | flake8 --diff`` and/or landscape.io linting service.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.

Please don't hesitate to ask for help if you're unsure of how to accomplish any of the above. You may delete all of these instructions except for the list above.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

### CODE REVIEW

**Comment by wholmgren** [comment]
Thanks! I like the conservative approach of only modifying values that are known to be problematic. 

It may be worthwhile to explicitly define the ``atol`` parameter, even if it's equal to the default. 

Can you add a simple test function to ``pvlib/test/test_solarposition.py`` that would have failed without these changes? The code in [your comment](https://github.com/pvlib/pvlib-python/issues/420#issuecomment-366271738) could work if you replace the print statements with ``assert not bad_azi.any()``. The new test should include a comment along the lines of ``# GitHub issue 420`` (I often forget to do this, but it does help future readers).

Finally, please add your name and a comment to the ``docs/sphinx/source/whatsnew/0.5.2.rst`` file.

**Comment by adriesse** [comment]
Hi Will, could you give some additional hints on the flake8 action item?

**Comment by wholmgren** [comment]
The command checks the changes that you've made for PEP8 style issues. landscape.io is supposed to do this for us, but it's been unreliable for awhile now. I haven't gotten around to setting up a new service. Stickler-CI is showing up in a few other projects that I follow, so maybe we'll try that.

In any case, for this pull request, flake8 will complain about how you've used whitespace to make the lines more similar. However, I don't have a problem with the formatting you're using, so no need to worry about it.

If you're still interested in pursuing it, you can read the forking and the pep8 sections in the pandas contributing guide:

http://pandas.pydata.org/pandas-docs/stable/contributing.html

The forking section shows how to add the "upstream remote" (the main pvlib repository). The pep8 section talks a little more about the flake8 command. Note that you'll need to run the diff against "upstream/master" rather than your local master.

**Comment by adriesse** [comment]
Ok, I'll learn about flake8 another time.
The latest whatsnew file I see is for 0.4.5.  Which one of us starts a new one?

**Comment by wholmgren** [comment]
Your fork shows that you have the 0.5.2 whatsnew file https://github.com/adriesse/pvlib-python/blob/master/docs/sphinx/source/whatsnew/v0.5.2.rst

Maybe you're looking at a different local copy of the code?

**Comment by adriesse** [comment]
No, I see now that the file extension changed from txt to rst.

**Comment by wholmgren** [comment]
Looks good, thanks Anton!!

File: docs/sphinx/source/whatsnew/v0.5.2.rst [hunk 1]
```diff
@@ -17,11 +17,15 @@ Bug fixes
 * physicaliam now returns a Series if called with a Series as an
   argument. (:issue:`397`)
 * corrected docstring for irradiance.total_irrad (:issue: '423')
+* modified solar_azimuth_analytical to handle some borderline cases, thereby
+  avoiding the NaN values and/or warnings that were previously produced
+  (:issue: '420')
```
Comment (mikofski) [inline docs/sphinx/source/whatsnew/v0.5.2.rst]: pls change single quotes around #420 to backticks

**Comment by mikofski** [comment]
so cool, thanks Anton!

**Pull request merged**